fix: docker compose configs

mjkeaton opened the pull request:
### **User description**
- Fix docker config
- Cache rust dependencies
- Add docker health checks

- Handling shutdown signals

Before:
```
quote-service-1 exited with code 137
swap-service-1 exited with code 137
treasury-service-1 exited with code 137
key-service-1 exited with code 137
```

After:
```
quote-service-1 exited with code 0
swap-service-1 exited with code 0
treasury-service-1 exited with code 0
key-service-1 exited with code 0
```


___

### **PR Type**
Enhancement, Bug fix, Configuration changes


___

### **Description**
- Added graceful shutdown handling for all services.

- Improved Docker configurations with health checks and dependency caching.

- Renamed configuration fields for consistency across services.

- Updated dependencies and added `tokio::signal` for signal handling.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>11 files</summary><table>
<tr>
  <td><strong>lib.rs</strong><dd><code>Renamed configuration fields for consistency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-3f8809d40f9e828f899b297d0eed35852796e26fa96f3c629ac0f4a6dc079589">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.rs</strong><dd><code>Added graceful shutdown signal handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-7ba54acb93cc4589ba66b78ffda6d81df6b37c6425ef914204c83fa374faeaec">+28/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>keys.rs</strong><dd><code>Renamed `base` to `base_url` in configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-d24b03325dacb5f2241120b3f10ef53990566604d941fe8a1dfd7a4ee08098ed">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.rs</strong><dd><code>Added graceful shutdown signal handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-2315d2c5dd613d7bc2bbdcea2b1512baa28f4bb7d62d0de18e1b50a1cc2a2ffa">+27/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lib.rs</strong><dd><code>Renamed configuration fields for clarity</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-d6043a0bd0e59231f714995c9050c95248343410c6e4944cd2d027c37ded0b06">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.rs</strong><dd><code>Added graceful shutdown signal handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-1740914fe229f4c6fd2faa6312452b6f9d3b4b6af729c5d9f8a921b0afff6858">+28/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.rs</strong><dd><code>Added graceful shutdown signal handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-ce3b84223bbd3d0908da85d994cb4a235eb6f85bed167479654f1f30c2b4268e">+27/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Dockerfile</strong><dd><code>Cached dependencies and added `curl` for health checks</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-e41f504f8018da13313791e0a395bed7c473cf27630e8a5820389b0cc091bf7c">+44/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Dockerfile</strong><dd><code>Cached dependencies and added `curl` for health checks</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-769970cbc6b811b1e568307ff3e1c048d115a7fc3491db9df233fcf33088e89f">+44/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Dockerfile</strong><dd><code>Cached dependencies and added `curl` for health checks</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-65f2806367f23577b1d770950430da30c7480b7ad53c0a932fb31c519c913f5c">+44/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Dockerfile</strong><dd><code>Cached dependencies and added `curl` for health checks</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-afca3a78f25cfc156da1818041a102a2acd0f955994b3a7f009f0f89cefc4a56">+44/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>Cargo.toml</strong><dd><code>Added `tokio::signal` feature for shutdown handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>docker-compose.yml</strong><dd><code>Added health checks and updated service dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3">+17/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>config.toml</strong><dd><code>Added `quotekeys` configuration section</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-ad1172e87dd1b09e7745ee6029056d0ced2d5322feea8e908065d0d1774f726a">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>config.toml</strong><dd><code>Renamed configuration section for clarity</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-623b7d2fd232079a81d400556040d9b31576e38463b1600363661c097cb3793e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>config.toml</strong><dd><code>Renamed `keys_cl` to `keys_client`</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/86/files#diff-4a22f2a5390b246e95833bc926a81eea6ca5ab706b0b0ceb46d5e0a24b821e43">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

mjkeaton committed:
chore: fix docker compose configs

mjkeaton committed:
chore: rename base to base_url for consistency

codingpeanut157 commented:
my bad, I changed config parameters names without changing the config files
good catch

As a general suggestion: I think `base_url` makes sense. Naming the parameter as simply `base` would make readers raise an eyebrow and a big :question:  on top of their heads.

`keys_cfg` on the other hand seems a bit redundant. I mean, the file is named config.toml, so I guess readers can already recon we are in the context of a configuration file.

mjkeaton committed:
docker: cache downloaded dependencies

mjkeaton committed:
docker: install curl for health checks

mjkeaton committed:
docker: add health check for key and quote services

mjkeaton committed:
docker: change docker data dir

mjkeaton committed:
docker: wait for key service to be healthy

mjkeaton commented:
> my bad, I changed config parameters names without changing the config files good catch

:+1: 

> 
> As a general suggestion: I think `base_url` makes sense. Naming the parameter as simply `base` would make readers raise an eyebrow and a big ❓ on top of their heads.

Agree. Changed to `base_url`.

> 
> `keys_cfg` on the other hand seems a bit redundant. I mean, the file is named config.toml, so I guess readers can already recon we are in the context of a configuration file.

Noted. Will be changed!

mjkeaton committed:
chore: handle shutdown signal

mjkeaton committed:
chore: remove _cfg postfix

mjkeaton committed:
chore: keys_cl to keys_client

codingpeanut157 reviewed (approved):
LGTM

waiting for cargo-chef to see how much boilerplate we can spare ourselves from  :smile:

codingpeanut157 committed:
[bcr-wdc-treasury-service] fix the config file

mjkeaton committed:
chore: git and docker ignore; treasury config

mjkeaton committed:
chore: docker mount data dir in treasury
